javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -1239,12 +1239,11 @@ private void doExtract(FileExtractor extractor, Path file, ExtractorState state)
         String msg = "A parse error occurred: " + StringUtil.quoteWithBackticks(err.getMessage().trim())
             + ". Check the syntax of the file. If the file is invalid, correct the error or [exclude](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/customizing-code-scanning) the file from analysis.";
         // file, relative to the source root
-        String relativeFilePath = null;
+        DiagnosticLocation.Builder builder = DiagnosticLocation.builder();
         if (file.startsWith(LGTM_SRC)) {
-          relativeFilePath = file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString();
+          builder = builder.setFile(file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString());
```

Comment:
Should we skip diagnostics, or perhaps just diagnostic locations, for files outside the source root?  I'm not sure how useful it is to have a location with a range but without a file path.